src/commentsModal/modal.js

```diff
@@ -0,0 +1,127 @@
+  const commentInput = document.createElement('textarea');
+  commentInput.id = 'comment-input';
+  commentInput.required = true;
+  const submitButton = document.createElement('button');
+  submitButton.type = 'submit';
+  submitButton.innerText = 'Comment';
+
+  // Append elements to create the modal structure
+  modalContent.appendChild(closeButton);
+  modalContent.appendChild(modalImage);
+  modalContent.appendChild(modalTitle);
+  modalContent.appendChild(modalDetails);
+  commentsSection.appendChild(commentsHeader);
+  commentsSection.appendChild(commentsCounter);
+  commentsSection.appendChild(commentsList);
+  commentForm.appendChild(nameLabel);
+  commentForm.appendChild(nameInput);
+  commentForm.appendChild(commentLabel);
+  commentForm.appendChild(commentInput);
+  commentForm.appendChild(submitButton);
+  commentsSection.appendChild(commentForm);
+  modalContent.appendChild(commentsSection);
+  modalContainer.appendChild(modalContent);
+  modalOverlay.appendChild(modalContainer);
+  document.body.appendChild(modalOverlay);
+
+  // Set the modal content
+  modalImage.innerHTML = `<img src="${movie.image.medium}" alt="${movie.name}">`;
+  modalTitle.innerText = movie.name;
+  modalDetails.textContent = movie.summary;
+
+  // Function to update comments in the modal
+  const updateComments = async () => {
+    try {
+      const comments = await getComments(movie.id);
+      const commentsCounter = document.getElementById('comments-counter');
+      commentsCounter.textContent = `Comments (${comments.length})`;
+      commentsList.innerHTML = '';
+
+      comments.forEach((comment) => {
+        const listItem = document.createElement('li');
+        listItem.innerText = `${comment.username}: ${comment.comment}`;
+        commentsList.appendChild(listItem);
+      });
+    } catch (error) {
+      // console.log('Error:', error.message);
+    }
+  };
+
+  // Close modal when clicking the close button or outside the modal
+  closeButton.addEventListener('click', () => {
+    modalOverlay.style.display = 'none';
+  });
+  modalOverlay.addEventListener('click', (event) => {
+    if (event.target === modalOverlay) {
+      modalOverlay.style.display = 'none';
+    }
+  });
+
+  // Prevent closing the modal when clicking inside the modal content
+  modalContent.addEventListener('click', (event) => {
+    event.stopPropagation();
+  });
+
+  // Prevent form submission and handle comment submission
+  commentForm.addEventListener('submit', async (event) => {
+    event.preventDefault();
+    const username = nameInput.value;
+    const comment = commentInput.value;
+
+    try {
+      await createComment(movie.id, username, comment);
+      nameInput.value = '';
+      commentInput.value = '';
+      await updateComments();
+    } catch (error) {
+    //   console.log('Error:', error.message);
+    }
+  });
+
+  // Display the modal
+  modalOverlay.style.display = 'block';
+
+  // Initialize comments
+  updateComments();
```

Comment:
- [ ] There's also a problem with the counter of your comments as it's not showing up:

![Screenshot from 2023-07-08 12-03-04](https://github.com/geekelo/kingflix/assets/77069011/3d4b18b9-ee3f-438d-b704-496c5945b135)

It looks like it shows up only the first time the popup opens and then If you try to open a new one it disappears. Please consider looking into this! 🤓 🔧